tests/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Rules/MiscellaneousRuleTests.cs

```diff
@@ -2,15 +2,246 @@
+        }
+
+        private void VerifyLocalizationInDesignTimeTargets()
+        {
+            VerifyLocalizationInProjectFile();
+            // For each DesignTime.targets file, check that the rule files with BrowseObject context are localized.
+
+            string managedDesignTimeTargets = Path.Combine(RepoUtil.FindRepoRootPath(), "src", "Microsoft.VisualStudio.ProjectSystem.Managed", "ProjectSystem", "DesignTimeTargets", "Microsoft.Managed.DesignTime.targets");
+            string cSharpDesignTimeTargets = Path.Combine(RepoUtil.FindRepoRootPath(), "src", "Microsoft.VisualStudio.ProjectSystem.Managed", "ProjectSystem", "DesignTimeTargets", "Microsoft.CSharp.DesignTime.targets");
+            string fSharpDesignTimeTargets = Path.Combine(RepoUtil.FindRepoRootPath(), "src", "Microsoft.VisualStudio.ProjectSystem.Managed", "ProjectSystem", "DesignTimeTargets", "Microsoft.FSharp.DesignTime.targets");
+            string visualBasicDesignTimeTargets = Path.Combine(RepoUtil.FindRepoRootPath(), "src", "Microsoft.VisualStudio.ProjectSystem.Managed", "ProjectSystem", "DesignTimeTargets", "Microsoft.VisualBasic.DesignTime.targets");
+
+            BrowseObjectsShouldBeLocalized(managedDesignTimeTargets);
+            BrowseObjectsShouldBeLocalized(cSharpDesignTimeTargets);
+            BrowseObjectsShouldBeLocalized(fSharpDesignTimeTargets);
+            BrowseObjectsShouldBeLocalized(visualBasicDesignTimeTargets);
+
+            if (XamlRules.Any())
+            {
+                _output.WriteLine($"There are more rules in the project file than in the DesignTime.targets files. Rules left: {XamlRules.Values}");
+                LocalizationIsClean = false;
+            }
+
+            Assert.True(LocalizationIsClean, "See test output for details.");
+        }
+
+        private void BrowseObjectsShouldBeLocalized(string designTimeTargetFile)
+        {
+            XElement? root = LoadXamlRule(designTimeTargetFile, out XmlNamespaceManager? namespaceManager);
+            var propertyPageSchemas = root.XPathSelectElements(@"/project:Project/project:ItemGroup/project:PropertyPageSchema", namespaceManager);
+
+            foreach (XElement element in propertyPageSchemas)
+            {
+                string? context = element.XPathSelectElement("./project:Context", namespaceManager)?.Value;
+                string resourceDirectory = element.Attribute("Include").Value;
+
+                Regex rx = new Regex(@"\).*\.xaml", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+                Match match = rx.Match(resourceDirectory);
+                string ruleName = match.Value.Substring(1);
+                bool alreadySeen = false;
+
+                // To do: How to know the full path. Some are under Rules, Rules/Items, Rules/PropertyPages
+                //string fullPath = Path.Combine(RepoUtil.FindRepoRootPath(), "src", "Microsoft.VisualStudio.ProjectSystem.Managed", "ProjectSystem", "Rules", ruleName);
```

Comment:
Can this go the other way, so you start with the path, then look up the file name? The other way around may be harder.

I expect it's an error to have two rules in different folders but with the same name, as they seem to be merged during build. If you loop through the files first, you could also verify there are no duplicate file names. If that'd be a headache to refactor to, I wouldn't worry.